### ISSUE TITLE
Cant have Reply-To filter working

### ISSUE THREAD

**Issue by schoubi** [osTicket/osTicket#5851]
### Description

Hi,

I've read the previous resolved issues about Reply-To filter, but infortunately i can't make it worked...

Like others, we have a web form that use a "From: Dont Reply <webadmin@domain.com>" and "To: osticket@domain.com". The reply-to is "Reply-To: "Bill Baroud" <bill.baroud@domain.com>".

The filter have this conditions :

Rules:
   - Match All : 
   - User/Email equals nobody@domain.com
Filter action :
  - Use Reply-To Email :heavy_check_mark: 

When we use the form, the user in the ticket list is always «Dont Reply»....

I tried many Rules, even one that catch all emails, without success...

Can you give me a hand ?

### Versions

osTicket Version         |	v1.14.5
Web Server Software |   Apache/2.4.6 (CentOS) OpenSSL/1.0.2k-fips mod_fcgid/2.3.9 PHP/5.6.40 mod_wsgi/3.4 Python/2.7.5
MySQL Version          |   5.5.68
PHP Version               |    5.6.40

Email Headers :

```
Return-Path: <webadmin@domain.com>
Received: from smtp.domain.com
[...]
Date: Tue, 30 Mar 2021 14:08:38 +0200
Message-Id: <2021031208.12UC8cL1019257@domain.com>
X-Authentication-Warning: www.domain.com: apache set sender to webadmin@domain.com using -f
To: osticket@domain.com
Subject: document no 4636
X-PHP-Originating-Script: 0:system.mail.inc
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8; format=flowed; delsp=yes
Content-Transfer-Encoding: 8Bit
X-Mailer: Drupal Webform (PHP/7.3.25)
Sender: webadmin@domain.com
From: "Dont Reply" <webadmin@domain.com>
Reply-To: "Bill Baroud" <bill.baroud@domain.com>
```

**Comment by protich** [comment]
This is a known issue - I would suggest you adjust your Drupal script to use User's name in From address as shown below (for now until we release a fix).

```
From: "Bill Baroud" <webadmin@domain.com>
```

**Comment by schoubi** [comment]
Thanks for your quick answer, we will try that.

This will create a user "Bill Baroud" in osticket with his proper mail address ?

**Comment by ChrisTG742** [comment]
Same here. Unfortunately I can't solve this by changing the "From"-address.
We use Power Automate to generate support-emails and later support-tickets in osTicket.
The support-email account does not have the right to send on behalf of every user in our directory.
Why did the suggested fix here never get included in the codebase? https://forum.osticket.com/d/494-email-header-reply-to-is-ignored/8
I think respecting a "reply-to" if present in the mail-header is the cleanest solution for this problem.

**Comment by protich** [comment]
@ChrisTG742 

 I'm sure from outside looking in it looks like the cleanest solution - I understand it addresses your issue, but let me explain why the suggested solution is shortsighted  and will break ticket filters. For starters it is only implemented for mail fetching - you'll need to address the same issue for piped emails.

osTicket defers switching to Reply-To address to the Ticket filters by design.  This is important because filters can be applied to the original `From` as well as to `Reply-To` address. Take the following example. 

```
To: osticket@domain.com
Subject: Web Ticket
From: "Dont Reply" <wordpress@domain.com>
Reply-To: "John Doe" <john.doe@example.com>
```

Technically you can setup a filter to route tickets sent From `wordpress@domain.com` to a particular department or assign to a team or an agent for example - while another filter can independently target `john.doe@example.com` etc. This is possible because osTicket re-triggers Ticket Filters when sender address changes.

Another equally important reason we opted not to blindly use Reply-To address by default is -  it can be used to send out spam, if incoming message is reflected back in an auto-response via %{thread.original} or full %{thread} variables. Spoofing `From` address is getting harder due to mail servers checking SPF & DKIM - but spammers can get around it by sending an email from legitimate email address e.g `spammer@gmail.com` and setting Reply-To header to `target@domain.com` -  now your osTicket installation is sending out SPAM.

To recap - Yes, osTicket already allows you to selectively setup a filter to use Reply-To address by targeting all email channels  or a particular one. The issue is - to use above header example - we don't currently use name on Reply-To (if available), we default to using name on From address which most mailers set correctly. That's the bug that needs to be fixed.

PS: It took me longer to type up this response than it would take to fix the bug, but I digress.

**Comment by ChrisTG742** [comment]
@protich Thanks for the detailed explanation! You're right, this is an issue if you use the system for customer support. In our case we just use in internally for supporting our empolyees. 
Still it would be great to have an option to enable osTicket to process a "reply-to" header in incoming emails the way my patch does. Using a filter rule for that is like a sledgehammer to crack a nut IMHO.

What about a patch that adds a new option at "Emails/Emails" or maybe "Emails/Settings" where one can optionally enable the processing of "reply-to" for that email address? Would you consider accepting the patch if I pimp it up to this?

**Comment by protich** [comment]
Perhaps  we can consider such a change as a global setting - but it will create 2 ways of doing something and also kills ability for filters to target original FROM address. People will obviously check the box and complain that filters don't work anymore.

**Comment by ChrisTG742** [comment]
Ok, so all that's currently broken is, that the "name" is not extracted correcly from the "reply-to" header, is that correct?
Maybe you could give me a small hint here I can find it in the code then I will try to supply a patch that really addresses the issue described here.

**Comment by protich** [comment]
Actually nothing is broken. My team and I did some testing this morning for both email fetching and piping and it works as expected. Both Reply-To name and email address are picked up correctly. 

All you need to do is add a Ticket Filter as shown below.

<img width="946" alt="Screen Shot 2021-10-28 at 11 09 57 AM" src="https://user-images.githubusercontent.com/31067/139295377-fdf94e70-6942-4dc7-8cff-d4129ccc7b08.png">

Of course the Filter Rules can be as complex as you want - taking note of possible spam relay I mentioned earlier.

<img width="951" alt="Screen Shot 2021-10-28 at 11 11 24 AM" src="https://user-images.githubusercontent.com/31067/139295504-bd97f78e-0a88-40b4-ba8f-3c2c9f6c1bef.png">

What might be a little confusing about how osTicket handles email addresses and names  is; We deduplicate emails & names - an email can only be associated to one name and email addresses are unique across the system. **Meaning when a user creates a ticket for the first time (either via web or email) - the name associated with the email is set and cannot change because of a subsequent  new email or web ticket using the same email address but different name.** 

This is done to prevent anonymous name injection / change.  Agents or End User can of course update the name upon login.

So the verdict is - this is not a bug at all.
